core/domain/summary_services_test.py

```diff
@@ -879,3 +879,170 @@ def test_requiring_public_activities_to_be_public_succeeds(self):
                 feconf.ACTIVITY_TYPE_EXPLORATION, self.EXP_ID_0),
             activity_domain.ActivityReference(
                 feconf.ACTIVITY_TYPE_COLLECTION, self.COL_ID_2)])
+
+
+class CollectionNodeExplorationDictsTest(
+        exp_services_test.ExplorationServicesUnitTests):
+    """Test functions for getting exploration summary dicts."""
+
+    ALBERT_EMAIL = 'albert@example.com'
+    BOB_EMAIL = 'bob@example.com'
+    ALBERT_NAME = 'albert'
+    BOB_NAME = 'bob'
+
+    USER_C_NAME = 'c'
+    USER_D_NAME = 'd'
+    USER_C_EMAIL = 'c@example.com'
+    USER_D_EMAIL = 'd@example.com'
+
+    USER_C_PROFILE_PICTURE = 'c_profile_picture'
+
+    EXP_ID_1 = 'eid1'
+    EXP_ID_2 = 'eid2'
+    EXP_ID_3 = 'eid3'
+    EXP_ID_4 = 'eid4'
+    EXP_ID_5 = 'eid5'
+
+    EXPECTED_VERSION_1 = 4
+    EXPECTED_VERSION_2 = 2
+
+    def setUp(self):
+        """Populate the database of explorations and their summaries.
+
+        The sequence of events is:
+        - (1) Albert creates EXP_ID_1.
+        - (2) Bob edits the title of EXP_ID_1.
```

Comment:
I don't understand why most of this is relevant. The tests should be minimal, in order to test just the functionality you care about. Otherwise, if something goes wrong, it could be due to any number of things.

So, just populate enough explorations so that get_collection_node_exploration_dicts() has something to work with, and just test that various inputs to that give the correct outputs.